sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsChangeTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+            var tagsForUpdate = TestDataGenerator.GenerateTags();
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: tagsForUpdate));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEquivalentTo(tagsForUpdate);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsRemoveTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName, tags: TestDataGenerator.GenerateTags());
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: TestDataGenerator.GenerateDnsResolver(tags: null));
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded);
+            updatedDnsResolver.Tags.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void DeleteDnsResolver_ResolverNotExists_ExpectNoError()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+
+            Action deleteDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.Delete(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName);
+
+            deleteDnsResolverAction.Should().NotThrow();
+        }
+
+        [Fact]
+        public void DeleteDnsResolver_ResolverExists_ExpectDeleted()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+
+            Action deleteResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.Delete(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name);
```

Comment:
This is not really verifying that the resolver is deleted.

Reply:
Added List Dns Resolvers to validate.